src/lib/db.ts

```diff
@@ -302,8 +316,10 @@ export class DB {
     });
   }
 
-  public openDocument(userId: string, documentKey: string) {
-    const {user} = this.stores;
+  public openDocument(userId: string, offeringDoc: DBOfferingUserSectionDocument) {
```

Comment:
This function was intended to work for any future user document, like learning logs - that is why it is passed a document key.  Could you update the caller so it can go back to receiving a document key?

Reply:
By update the caller I mean move the monitoring to another function that accepts a `DocumentModel` object.

Reply:
Makes sense, more modular now